File: src/python/linked_list/remove_duplicates_from_sorted_list.py
from python.tools.node import ListNode


class Solution:

    def deleteDuplicates(self, head: ListNode) -> ListNode:
        """
        100ms 21.85%
        13.3MB 17.24%
        :param head:
        :return:
        """
        if not head:
            return head

        fast = slow = head

        while fast:
            if fast.val != slow.val:
                slow.next = fast
                slow = slow.next
            fast = fast.next

        slow.next = fast

        return head

File: leetcode-java/src/classify/list/RemoveDuplicatesFromSortedList.java
package classify.list;

import tools.ListNode;

public class RemoveDuplicatesFromSortedList {

    /**
     * 0ms 100.00%
     * 35.5MB 99.92%
     */
    public ListNode deleteDuplicates(ListNode head) {
        ListNode cur = head;

        while (cur != null) {
            if (cur.next == null) {
                return head;
            } else if (cur.next.val == cur.val) {
                cur.next = cur.next.next;
            } else {
                cur = cur.next;
            }
        }

        return null;
    }

    /**
     * 0ms      100.00%
     * 41MB     54.09%
     */
    public ListNode deleteDuplicates2(ListNode head) {
        ListNode cur = head;

        while (cur != null && cur.next != null) {
            if (cur.next.val == cur.val) {
                cur.next = cur.next.next;
            } else {
                cur = cur.next;
            }
        }

        return head;
    }
}
